### ISSUE TITLE
gif support

### ISSUE THREAD

**Issue by d7eeem** [wader/ydls#382]
i forked the project and i tried to add gif support to JSON config and when i try to build it it give an error 
```
ERROR: failed to solve: process "/bin/sh -c CONFIG=/src/ydls.json   TEST_EXTERNAL=1   go test -timeout=30m -buildvcs=false -v -cover $([ \"$TARGETARCH\" = \"amd64\" ] && echo -race) ./..." did not complete successfully: exit code: 1
```

**Comment by wader** [comment]
Hey, yes i think that should be possible. Could you post your changed files somewhere?

BTW if you only want to change the config i think you can volume mount a file at /etc/ydls.json (`-v /path/on/host/ydls.json:/etc/ydls.json`), that way you don't need to rebuild the image.

**Comment by d7eeem** [comment]
i need to build it because i don't know how to do it the other way in unriad

**Comment by wader** [comment]
I see, but could you post your config so i can try to reproduce?

**Comment by wader** [comment]
This config seems to work:
```json
    "gif": {
      "Formats": ["gif"],
      "Streams": [
        { "Required": true, "Specifier": "v:0", "Codecs": ["gif"] }
      ],
      "Ext": "gif",
      "MIMEType": "image/gif"
    }
```

I think the reason the test fails with this config is that ffmpeg's gif demuxer does not like reading a streamed gif as input (seems is tried to seek and gets IO error) and currently the tests require streaming. I guess a workaround could be to write to a intermediate temp file. This probably also means that transcoding from a gif file will fail for the same reason.

Another issue might be that the palette used by the output gif will not be very optimal. To improve that i think some kind of "2 pass" is required which is not easy to do at the moment.

**Comment by wader** [comment]
BTW you can skip most tests by uncomment or remove these lines https://github.com/wader/ydls/blob/master/Dockerfile#L54-L57

**Comment by wader** [comment]
Seems to be fixed in ffmpeg master https://github.com/FFmpeg/FFmpeg/commit/c6b6356635f598b095606cd126f31bc6ab916225 so hopefully will be fixed in ffmpeg 6.1
```sh
$ ./ffmpeg -v quiet -f lavfi -i testsrc -t 1s -f gif pipe:1 | ./ffprobe -hide_banner -i pipe:0
Input #0, gif, from 'pipe:0':
  Duration: N/A, start: 0.000000, bitrate: N/A
  Stream #0:0: Video: gif, bgra, 320x240 [SAR 64:64 DAR 4:3], 25 fps, 25 tbr, 100 tbn
```

**Comment by wader** [comment]
Should work now, but beware of no image resize and probably not very optimal color palette usage
